### DISCUSSION TITLE
Problem when package PCL

### DISCUSSION THREAD

**Discussion by bladesaber** [pybind/pybind11#3940] [Q&A]
Hello, everyone. 
     I want to make a PCL python wrapper, but after I build the .so and want to import it in python. It throws an error: dynamic module does not define module export function (PyInit_pcl_lib).

     CMakeLists.txt below:
`cmake_minimum_required(VERSION 3.0.2)`
`project(pybind_test)`
`set(PACKAGE_TOP_DIR ${CMAKE_CURRENT_SOURCE_DIR})`
`find_package(PCL 1.8 REQUIRED)`
`add_subdirectory(cpp/pybind11)`
`include_directories(${PCL_INCLUDE_DIRS})`
`link_directories(${PCL_LIBRARY_DIRS})`
`add_definitions(${PCL_DEFINITIONS})`
`pybind11_add_module(pcl_lib cpp/src/pcl_lib.cpp)`
`target_link_libraries(pcl_lib PUBLIC pybind11::module ${PCL_LIBRARIES})`

    pcl_lib.cpp below

`#include <pybind11/pybind11.h>`
`int add(int i, int j) {
    return i + j;
}`
`PYBIND11_MODULE(pcl_lib, m){
    m.doc() = "pcl plugin";
    m.def("add", &add, "A function that adds two numbers");
}`

Looking forward to your response, thank you. ^_^

**Comment by RPFey** [comment]
Hi, I had the same problem as you, using pcl-1.8; I solved this by building my packaged using pcl-1.10. Good Luck!

**Comment by liuwenhai** [comment]
find_package(PCL 1.8 REQUIRED) uses its dependent libpython.so, which may be different from the python version you use.
a direct method is replacing  find_package(PCL 1.8 REQUIRED) with the following code,

include_directories(/where/your/pcl/install/include/pcl-1.8  )
file(GLOB PCL_LIBRARIES "/where/your/pcl/install/lib/libpcl*.so*")
find_package(Boost REQUIRED filesystem thread)
include_directories(${Boost_INCLUDE_DIRS})
find_package(Eigen3 REQUIRED)
include_directories(${EIGEN3_INCLUDE_DIR})
target_link_libraries(shot PRIVATE ${PCL_LIBRARIES} ${Boost_LIBRARIES})
